src/daemon/control.rs

```diff
@@ -1699,4 +1707,492 @@ mod test {
+        tx.finalize(&revaultd.secp_ctx).unwrap();
+        presigned_tx_sighash(&tx, SigHashType::All).unwrap_err();
+
+        // A RevaultTransaction with the wrong SigHashType
+        let tx = SpendTransaction::from_psbt_str("cHNidP8BAKgCAAAAAWQYLaJLKOGr2VUPYwlZz5UStWUr7SlxGT+K8S8ubTM1AAAAAAADAAAAA6BCAAAAAAAAIgAg6eJKGnlvCyzkIYItHMBwQDkooqkNDi9RvlVWyl+rYTGA8PoCAAAAABYAFCgf4ZTSb/CpYwWa+Wp0yRMIJoVYf4D4AgAAAAAiACB18mkXdMgWd4MYRrAoIgDiiLLFlxC1j3Qxg9SSVQfbxQAAAAAAAQErybfzBQAAAAAiACCoQFnKMvBMFQZ1py8aLwPpRU6pUCx/LKGU9VMp//zoOwEDBAEAAAABBaghAgKTOrEDfq0KpKeFjG1J1nBeH7O8X2awCRive58A7NUmrFGHZHapFHKpXyKvmhuuuFL5qVJy+MIdmPJkiKxrdqkUtsmtuJyMk3Jsg+KhtdlHidd7lWGIrGyTUodnUiEDD2S5Iq7i/Vl/EEvGyztnDxyixsSbEHGhpsAQV12U/lohAqvkdbGZ7D1i+ldvruFqM0/bhv+ybc51vs66rt8yisP+Uq9TsmgAAQElIQK1323qfhEH7yMqxloxMQOfxx7VhZrl5zso8JRdkhBfH6xRhwAAAQFHUiECWC3tv0T0ZWTl2M2wZ1NtYOvjTNHRgBz/Ubv516wom0MhA0cE3stVtaqI/9HvXQY2YkjBMU4ZZVETb/FOq4u6SkkOUq4A").unwrap();
+        presigned_tx_sighash(&tx, SigHashType::Single).unwrap_err();
+
+        // And finally, a proper RevaultTransaction :)
+        let tx = SpendTransaction::from_psbt_str("cHNidP8BAKgCAAAAAWQYLaJLKOGr2VUPYwlZz5UStWUr7SlxGT+K8S8ubTM1AAAAAAADAAAAA6BCAAAAAAAAIgAg6eJKGnlvCyzkIYItHMBwQDkooqkNDi9RvlVWyl+rYTGA8PoCAAAAABYAFCgf4ZTSb/CpYwWa+Wp0yRMIJoVYf4D4AgAAAAAiACB18mkXdMgWd4MYRrAoIgDiiLLFlxC1j3Qxg9SSVQfbxQAAAAAAAQErybfzBQAAAAAiACCoQFnKMvBMFQZ1py8aLwPpRU6pUCx/LKGU9VMp//zoOwEDBAEAAAABBaghAgKTOrEDfq0KpKeFjG1J1nBeH7O8X2awCRive58A7NUmrFGHZHapFHKpXyKvmhuuuFL5qVJy+MIdmPJkiKxrdqkUtsmtuJyMk3Jsg+KhtdlHidd7lWGIrGyTUodnUiEDD2S5Iq7i/Vl/EEvGyztnDxyixsSbEHGhpsAQV12U/lohAqvkdbGZ7D1i+ldvruFqM0/bhv+ybc51vs66rt8yisP+Uq9TsmgAAQElIQK1323qfhEH7yMqxloxMQOfxx7VhZrl5zso8JRdkhBfH6xRhwAAAQFHUiECWC3tv0T0ZWTl2M2wZ1NtYOvjTNHRgBz/Ubv516wom0MhA0cE3stVtaqI/9HvXQY2YkjBMU4ZZVETb/FOq4u6SkkOUq4A").unwrap();
```

Comment:
So weird to test validity of a routine for *presigned* transactions using a Spend transaction :sweat_smile:. We should probably aim for stronger types in the future (instead of taking a `&impl RevaultTransaction`).

Reply:
I agree, we could have a supertrait `PresignedTransaction: RevaultTransaction` :)

Reply:
It wasn't a request to change it, FWIW: i like that it exposed it. Don't change it again with this comment though :)

Agree for the supertrait. This, or an enum.